runtime/calamari/tests/common/mod.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright 2020-2022 Manta Network.
+// This file is part of Manta.
+//
+// Manta is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Manta is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Manta.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Calamari Parachain Integration Tests.
+
+pub use calamari_runtime::{
+	currency::KMA, Call, CollatorSelection, Democracy, Event, Origin, Runtime, Scheduler, Session,
+	System, TransactionPayment,
+};
+use frame_support::{
+	traits::{GenesisBuild, OnFinalize, OnInitialize},
+	weights::{DispatchInfo, Weight},
+};
+use manta_primitives::{
+	helpers::{get_account_id_from_seed, get_collator_keys_from_seed},
+	AccountId, AuraId, Balance,
+};
+use sp_core::sr25519;
+
+pub const BOND_AMOUNT: Balance = 1_000 * KMA;
+pub const INITIAL_BALANCE: Balance = 1_000_000_000_000 * KMA;
+
+/// create a transaction info struct from weight. Handy to avoid building the whole struct.
+pub fn info_from_weight(w: Weight) -> DispatchInfo {
+	// pays_fee: Pays::Yes -- class: DispatchClass::Normal
+	DispatchInfo {
+		weight: w,
+		..Default::default()
+	}
+}
+
+/// Utility function that advances the chain to the desired block number.
+pub fn run_to_block(n: u32) {
+	while System::block_number() < n {
+		System::on_finalize(System::block_number());
+		System::set_block_number(System::block_number() + 1);
+		System::on_initialize(System::block_number());
+		Session::on_initialize(System::block_number());
+		Scheduler::on_initialize(System::block_number());
+		Democracy::on_initialize(System::block_number());
+	}
+}
+
+pub fn last_event() -> Event {
+	System::events().pop().expect("Event expected").event
+}
+
+pub struct ExtBuilder {
```

Comment:
Wouldn't this be better placed in a mock.rs ?

Reply:
Yeah makes sense.